buildSrc/src/main/minimumRuntime/org/elasticsearch/gradle/LoggedExec.java

```diff
@@ -19,21 +26,52 @@
 @SuppressWarnings("unchecked")
 public class LoggedExec extends Exec {
 
+    // helper to allow lambda creation using Files methods
+    private interface ThrowingStringSupplier {
+        String get() throws Exception;
+    }
+
+    private Property<Boolean> useFileBuffer;
+    
     public LoggedExec() {
-        ByteArrayOutputStream output = new ByteArrayOutputStream();
-        ByteArrayOutputStream error = new ByteArrayOutputStream();
+        this.useFileBuffer = getProject().getObjects().property(Boolean.class);
+        this.useFileBuffer.set(false); // default to in memory
         if (getLogger().isInfoEnabled() == false) {
-            setStandardOutput(output);
-            setErrorOutput(error);
-            setIgnoreExitValue(true);
-            doLast((unused) -> {
+            getProject().afterEvaluate(p -> {
```

Comment:
I'd actually prefer putting this in a `doFirst()` than `afterEvaluate`. At least then we can avoid paying this cost at config time and we don't really care about tracking these things as inputs for UP-TO-DATE checking.

Even better would be providing some kind of `FileOutputStream` implementation that we can provide a `Provider<File>` to and would defer opening the file until the first write.